Advanced_Projects/Excel to CSV/excel_to_csv.py

```
@@ -0,0 +1,193 @@
+
+        self.convert_btn.config(state='disabled')
+        self.progress.configure(mode='determinate', value=0)
+        self.status_label.config(text="Converting...", foreground="#cccccc")
+
+        def convert():
+            try:
+                for result in excel_to_csv(input_file, output_folder):
+                    if isinstance(result, tuple):
+                        percent, msg = result
+                        if percent == -1:
+                            self.root.after(0, lambda: self.finish_conversion(False, msg))
```

Ruff (0.11.9)

Function definition does not bind loop variable `msg`

(B023)